src/components/common/DataTabel/Tabel.tsx

```diff
@@ -0,0 +1,99 @@
+"use client";
+
+import React from "react";
+import {
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { Member } from "@/types";
+import useGroupedData from "@/hooks/use-group-data";
+
+type DataTableProps = {
+  data: Member[];
+  columns: any;
+  isGroupData: boolean;
+};
+
+export const DataTabel = ({ data, columns, isGroupData }: DataTableProps) => {
+  const table = useReactTable({
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+  });
+
+  const groupedData = useGroupedData(data, "shift");
```

Comment:
This is wrong. This makes it static that every table that has a group header would be bind to `shift` property.

Reply:
Now I have made it dynamic, allowing us to group by any specified field.